### PR TITLE
FIXED | Schematics to work with angular 12

### PR DESCRIPTION
**Pull request by stones**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/ngneat/spectator/blob/master/CONTRIBUTING.md#commit
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

```
[x] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] Other... Please describe:
```

## What is the current behavior?
The schematics don't work due to additional properties being passed

Issue Number: 460

## What is the new behavior?
Removed deprecated options from schema.json and defaults.

## Does this PR introduce a breaking change?
```
[ ] Yes
[x ] No
```

## Other information

I tried to follow what was done in the Akita repo as suggested but I couldn't add properties to the existing schema and I thought that working was better than perfect. 

If there is something I am missing I would love to learn.

### CODE REVIEW

File: projects/spectator/schematics/src/spectator/index.ts [hunk 1]
```diff
@@ -23,7 +23,6 @@ export function spectatorComponentSchematic(options: ComponentOptions): Rule {
     externalSchematic('@schematics/angular', 'component', {
       ...omit(options, ['jest', 'withHost', 'withCustomHost']),
       skipTests: true,
-      spec: false,
```
Comment (NetanelBasal) [inline projects/spectator/schematics/src/spectator/index.ts:26]: why did you remove it?

**Comment by stones** [comment]
It was a deprecated flag that was removed in 12.

It was the same as the flag removed from line 92 is this file:

https://github.com/ngneat/spectator/pull/466/commits/5bb6357aada31c55c353481310844baf18457c96#diff-71c3a62fbfbecc2138567eed22f59504b8772037703f6b85692f56b400c53617
